src/transactions/test/InvokeHostFunctionTests.cpp

```diff
@@ -2065,17 +2072,21 @@ TEST_CASE("state archival", "[tx][soroban]")
         }
 
         // Contract instance and code should be expired
-        ledgerSeq = test.getLedgerSeq();
         auto const& contractKeys = client.getContract().getKeys();
-        REQUIRE(!test.isEntryLive(contractKeys[0], ledgerSeq));
-        REQUIRE(!test.isEntryLive(contractKeys[1], ledgerSeq));
+        REQUIRE(!test.isEntryLive(contractKeys[0], test.getLCLSeq()));
+        REQUIRE(!test.isEntryLive(contractKeys[1], test.getLCLSeq()));
+        // Wasm is created 1 ledger before code, so lives for 1 ledger less.
+        REQUIRE(test.getTTL(contractKeys[0]) ==
+                originalExpectedLiveUntilLedger - 1);
+        REQUIRE(test.getTTL(contractKeys[1]) ==
+                originalExpectedLiveUntilLedger);
 
         // Contract instance and code are expired, any TX should fail
         REQUIRE(client.put("temp", ContractDataDurability::TEMPORARY, 0) ==
                 INVOKE_HOST_FUNCTION_ENTRY_ARCHIVED);
 
         auto newExpectedLiveUntilLedger =
-            stateArchivalSettings.minPersistentTTL + ledgerSeq - 1;
+            (test.getLCLSeq() + 1) + stateArchivalSettings.minPersistentTTL - 1;
```

Comment:
Comment here would be helpful, as well as getting rid of the +1 followed by -1.

Reply:
Moved the code around to follow the usual pattern of LCL + min TTL - 1, so hopefully it's more clear now.